pkg/controller/kanarydeployment/strategies/scale/static.go

```diff
@@ -62,7 +62,7 @@ func (s *staticImpl) Clear(kclient client.Client, reqLogger logr.Logger, kd *kan
 	return status, reconcile.Result{}, nil
 }
 
-func updateDeploymentReplicas(kclient client.StatusWriter, reqLogger logr.Logger, dep *appsv1beta1.Deployment, replicas int32) (reconcile.Result, error) {
+func updateDeploymentReplicas(kclient client.Client, reqLogger logr.Logger, dep *appsv1beta1.Deployment, replicas int32) (reconcile.Result, error) {
```

Comment:
```suggestion
func updateDeploymentReplicas(kclient client.StatusWriter, reqLogger logr.Logger, dep *appsv1beta1.Deployment, replicas int32) (reconcile.Result, error) {
```

Reply:
I looked again at the code and really we need a client to update the spec not the status.